### PR TITLE
HCX-6 Allow partial updates in hid.Hydra#update* methods

### PR DESCRIPTION
**Pull request by dolfinus**
Previously you had to pass all entity fields to along with updated ones.
But not now.

### CODE REVIEW

File: src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy [hunk 1]
```diff
@@ -443,56 +443,64 @@ trait Address {
...
+        vch_VC_CODE             : params.code,
+        vch_VC_ADDRESS          : params.rawAddress,
+        num_N_REGION_ID         : params.regionId,
+        vch_VC_FLAT             : params.flat,
+        num_N_FLOOR_NO          : params.floor,
+        ch_C_FL_MAIN            : encodeBool(params.isMain),
+        num_N_ADDR_STATE_ID     : params.stateId,
+        vch_VC_REM              : params.rem
+      ]
       if (this.version >= '5.1.2') {
```
Comment (sugarfree1) [inline src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy]: Don't do that, please

File: src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy [hunk 2]
```diff
@@ -443,56 +443,64 @@ trait Address {
       rem            : null,
       stateId        : getDefaultAddressStateId(),
       isMain         : null
-    ], input)
+    ]
     try {
-      logger.info("Putting address with params ${params}")
+      if (notEmpty(input.subjAddressId)) {
+        LinkedHashMap subjAddress = getSubjAddress(input.subjAddressId)
+        defaultParams += [
```
Comment (sugarfree1) [inline src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy]: They are not default params. This is current value.

File: src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy (hunk 1 above)
Comment (dolfinus) [reply src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy]: Don't do what?

File: src/org/camunda/latera/bss/connectors/hid/hydra/Customer.groovy [hunk 3]
```diff
@@ -213,26 +214,45 @@ trait Customer {
...
+        num_N_SUBJECT_ID      : params.customerId,
         num_N_FIRM_ID         : params.firmId,
         num_N_BASE_SUBJECT_ID : params.baseSubjectId,
         num_N_SUBJ_STATE_ID   : params.stateId,
         num_N_SUBJ_GROUP_ID   : params.groupId,
         vch_VC_CODE           : params.code,
         vch_VC_REM            : params.rem
       ]
-      if (params.resellerId) {
+      if (this.version > 5 && notEmpty(params.resellerId)) {
```
Comment (sugarfree1) [inline src/org/camunda/latera/bss/connectors/hid/hydra/Customer.groovy]: Maybe ```new Version```?

**Pull request merged**